lightning/src/ln/channelmanager.rs

```diff
@@ -3784,26 +3797,27 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	/// Provides a payment preimage in response to [`Event::PaymentReceived`], generating any
 	/// [`MessageSendEvent`]s needed to claim the payment.
 	///
+	/// Note that calling this method does *not* guarantee that the payment has been claimed. You
+	/// *must* wait for an [`Event::PaymentClaimed`] event.
```

Comment:
Perhaps this could be worded to indicate the event will be available after returning, if it is to be generated. Otherwise, "wait" seems to indicated it may just take some time.